File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task44_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <limits>

int main() {
    int num1, num2;

    while (true) {
        std::cout << "Enter first number: ";
        if (std::cin >> num1) {
            break;
        } else {
            std::cout << "Invalid input. Please enter a valid number.
";
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '
');
        }
    }

    while (true) {
        std::cout << "Enter second number: ";
        if (std::cin >> num2) {
            break;
        } else {
            std::cout << "Invalid input. Please enter a valid number.
";
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '
');
        }
    }

    std::cout << "Sum: " << num1 + num2 << std::endl;
    std::cout << "Difference: " << num1 - num2 << std::endl;
    std::cout << "Product: " << num1 * num2 << std::endl;
    if (num2 != 0) {
        std::cout << "Quotient: " << num1 / num2 << std::endl;
    } else {
        std::cout << "Cannot divide by zero." << std::endl;
    }

    return 0;
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task44_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.InputMismatchException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int num1, num2;

        while (true) {
            try {
                System.out.print("Enter first number: ");
                num1 = scanner.nextInt();
                break;
            } catch (InputMismatchException e) {
                System.out.println("Invalid input. Please enter a valid number.");
                scanner.next();
            }
        }

        while (true) {
            try {
                System.out.print("Enter second number: ");
                num2 = scanner.nextInt();
                break;
            } catch (InputMismatchException e) {
                System.out.println("Invalid input. Please enter a valid number.");
                scanner.next();
            }
        }

        System.out.println("Sum: " + (num1 + num2));
        System.out.println("Difference: " + (num1 - num2));
        System.out.println("Product: " + (num1 * num2));
        if (num2 != 0) {
            System.out.println("Quotient: " + (num1 / num2));
        } else {
            System.out.println("Cannot divide by zero.");
        }
    }
}